Src/main.c

```diff
@@ -103,13 +103,37 @@ int main(void)
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
+  //Initialize ADC and TIM
+  HAL_ADC_Init(&hadc);
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1);
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  //set servo PWM output relative to input voltage from potentiometer.
+  //potentiometer voltage from 0-3.3V, PWM modulation from 1-2ms on-time at 50Hz (period = 20ms)
+  
+  int Ch1 = 0; //no need to constantly re-initialize a variable.
   while (1)
   {
+      Ch1 = 0;
+      //ADC conversion:
+      double OnTime = HAL_ADC_GetValue(&hadc); //RAW ADC input
```

Comment:
Curious, why the variable name OnTime?

Reply:
Made sense to me as it is the thickness of the on part of the duty cycle.